support/azure/azure-kubernetes/extensions/troubleshoot-node-auto-provision.md

```diff
@@ -0,0 +1,265 @@
+---
+title: Troubleshoot the Node Auto-provisioning managed add-on
+description: Learn how to troubleshoot Node Auto-provisisioning in Azure Kubernetes Service (AKS).
+ms.service: azure-kubernetes-service
+ms.date: 09/05/2025
+editor: bsoghigian
+ms.reviewer: 
+#Customer intent: As an Azure Kubernetes Service user, I want to troubleshoot problems that involve Node Auto-provisioining managed add-on so that I can successfully provision, scale, and manage my nodes and workloads on Azure Kubernetes Service (AKS).
+ms.custom: sap:Extensions, Policies and Add-Ons
+---
+
+# Troubleshoot node auto provisioning (NAP) in Azure Kubernetes Service (AKS)
+
+This article discusses how to troubleshoot Node auto provisioning(NAP), a managed add-on based on the open source [Karpenter](https://karpenter.sh) project. NAP automatically provisions and manages nodes in response to pending pod pressure, and manages scaling events at the virtual machine, or node level.
+ When you enable Node Auto-provisioning, you might experience problems that are associated with the configuration of the infrastructure autoscaler. This article will help you troubleshoot errors and resolve common problems that affect NAP but aren't covered in the official Karpenter [FAQ][karpenter-faq] and [troubleshooting guide][karpenter-troubleshooting].
+
+## Prerequisites
+
+Ensure the following tools are installed and configured. They're used in the following sections.
+
+- [Azure CLI](/cli/azure/install-azure-cli). To install kubectl by using the [Azure CLI](/cli/azure/install-azure-cli), run the [az aks install-cli](/cli/azure/aks#az-aks-install-cli) command.
+- The Kubernetes [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) tool, a Kubernetes command-line client. This is available with the Azure CLI.
+- Confirm you have Node Auto-provisioning enabled on your cluster.
+
+## Common Issues
+
+### Nodes Not Being Removed
+
+**Symptoms**: Underutilized or empty nodes remain in the cluster longer than expected.
+
+**Debugging Steps**:
+
+1. **Check node utilization**:
+```azurecli-interactive
+kubectl top nodes
+kubectl describe node <node-name>
+```
+You can also use the open-source [AKS Node Viewer](https://github.com/Azure/aks-node-viewer) tool to visualize node usage.
+
+2. **Look for blocking pods**:
+```azurecli-interactive
+kubectl get pods --all-namespaces --field-selector spec.nodeName=<node-name>
+```
+
+3. **Check for disruption blocks**:
+```azurecli-interactive
+kubectl get events | grep -i "disruption\|consolidation"
+```
+
+**Common Causes**:
+- Pods without proper tolerations
+- DaemonSets preventing drain
+- Pod disruption budgets(PDBs) are not properly set
+- Nodes are marked with `do-not-disrupt` annotation
```

Comment:
Locks, maybe as well? Although not common lifecycle its something customers set explicitly.